instance/models/instance.py

```diff
@@ -447,6 +447,23 @@ def provision(self):
 
         Returns: (server, log)
         """
+        attempt_num = 1
+        provisioned = False
+        server, logs = None, []
+
+        while attempt_num <= self.attempts and not provisioned:
+            server, deploy_log, provisioned = self._provision_attempt()
```

Comment:
It would be nice to continue logging the attempt number before each attempt, as we do now.  I find this information useful (and I search for it in the web interface to find the next attempt).